File: ielts-vocab-master/Python/Flask/auth/session.py
import json

from flask import Blueprint, jsonify, make_response, session

from db import get_db
from remember import consume_remember_cookie

session_bp = Blueprint('session_check', __name__)


@session_bp.route('/session')
def check_session():
    conn = get_db()

    if 'user_id' not in session:
        placeholder = make_response()
        user_id = consume_remember_cookie(conn, placeholder)
        if not user_id:
            return jsonify({'loggedIn': False})
        session.clear()
        session['user_id'] = user_id
        extra_cookies = placeholder.headers.getlist('Set-Cookie')
    else:
        user_id = session['user_id']
        extra_cookies = []

    row = conn.execute(
        'SELECT id, email, name, username, picture FROM users WHERE id = ?', (user_id,)
    ).fetchone()

    if not row:
        session.pop('user_id', None)
        return jsonify({'loggedIn': False})

    resp = make_response(jsonify({
        'loggedIn':  True,
        'id':        row['id'],
        'email':     row['email'],
        'name':      row['name'],
        'username':  row['username'],
        'picture':   row['picture'],
    }))
    for cookie in extra_cookies:
        resp.headers.add('Set-Cookie', cookie)
    return resp

File: VocabLarry/PHP/auth/session.php
<?php
session_start();
header('Content-Type: application/json');
require __DIR__ . '/db.php';
require __DIR__ . '/remember.php';

$pdo = get_db();

if (!isset($_SESSION['user_id'])) {
    $userId = consume_remember_cookie($pdo);
    if (!$userId) {
        echo json_encode(['loggedIn' => false]);
        exit;
    }
    session_regenerate_id(true);
    $_SESSION['user_id'] = $userId;
}

$stmt = $pdo->prepare('SELECT id, email, name, username, picture FROM users WHERE id = ?');
$stmt->execute([$_SESSION['user_id']]);
$user = $stmt->fetch(PDO::FETCH_ASSOC);

if (!$user) {
    unset($_SESSION['user_id']);
    echo json_encode(['loggedIn' => false]);
    exit;
}

echo json_encode([
    'loggedIn' => true,
    'id'       => $user['id'],
    'email'    => $user['email'],
    'name'     => $user['name'],
    'username' => $user['username'],
    'picture'  => $user['picture'],
]);
